docs/it/protocolli-di-comunicazione.md

```diff
@@ -0,0 +1,93 @@
+Potremmo aver incontrato questo formato in diversi contesti, ad esempio quando si effettua una chiamata HTTP ad un server web, il server restituisce una risposta in formato JSON. Oppure in un file di configurazione.
+
+### Il formato XML
+
+Il formato XML (eXtensible Markup Language) è un formato testuale per la rappresentazione dei dati applicativi. Questo formato è basato sulla sintassi del linguaggio XML e permette di rappresentare dati in modo strutturato e leggibile. Questo formato è molto diffuso poiché permette di definire un formato personalizzato per la rappresentazione dei dati, usando una definizione di schema.
+Potremmo aver incontrato questo formato in diversi contesti, ad esempio in un servizio SOAP, in un file di configurazione o in un file SVG.
+
+### Il formato YAML
+
+Il formato YAML (YAML Ain't Markup Language) è un formato testuale per la rappresentazione dei dati applicativi. Questo formato è basato sulla sintassi del linguaggio YAML e permette di rappresentare dati in modo strutturato e leggibile, evitando la verbosità del formato XML e JSON.
+Potremmo aver incontrato questo formato in diversi contesti, ad esempio in un file di configurazione.
+
+### Il formato Protocol Buffers
+
+Il formato Protocol Buffers è un formato binario per la rappresentazione dei dati applicativi. Questo formato è basato sulla sintassi del linguaggio Protocol Buffers e permette di rappresentare dati in modo strutturato e veloce. Questo formato è molto diffuso poiché permette di definire un formato personalizzato per la rappresentazione dei dati, usando una definizione di schema. Permette di gestire in modo semplice e veloce la serializzazione e la deserializzazione dei dati anche in modo retrocompatibile.
+Potremmo aver incontrato questo formato in diversi contesti, ad esempio in un servizio gRPC, essendo un formato binario e quindi più veloce e più efficiente di un formato testuale.
+
+## Conclusioni
+
+In questo capitolo abbiamo visto i principali protocolli di comunicazione, come funzionano e in quali ambiti vengono utilizzati. Abbiamo visto che i protocolli di comunicazione sono un insieme di regole che definiscono come avviene la comunicazione tra due o più entità, in questo caso tra due o più computer. Queste regole definiscono come avviene la comunicazione, come vengono trasmesse le informazioni e come vengono interpretate.
+Non tutti i protocolli di comunicazione sono uguali: alcuni sono più diffusi di altri, alcuni più complessi di altri, alcuni più sicuri di altri, alcuni più performanti di altri, alcuni più adatti di altri a determinati scopi. È importante quindi scegliere il protocollo di comunicazione più adatto al contesto in cui si lavora.
```

Comment:
@allevo Qui ho il medesimo dubbio di [qui](https://github.com/Il-Libro-Open-Source/book/pull/196/files#diff-3fe0312258659ab0d7aa36e42decef10d95bf10d9632d334c2fc610e2d142517R10), questa continua ripetizione, durante la lettura della frase, ha qualcosa che non mi convince appieno.

Fermo restando che qualora fosse grammaticalmente corretta e risultasse buona in fase di lettura, la troverei comunque una ripetizione della frase di cui sopra.